File: android/app/src/main/kotlin/com/thisissandipp/one/contacts/models/PostalAddress.kt
package com.thisissandipp.one.contacts.models

data class PostalAddress(
    val formattedAddress: String? = "",
    val city: String? = "",
    val country: String? = "",
    val neighborhood: String? = "",
    val poBox: String? = "",
    val postCode: String? = "",
    val region: String? = "",
    val street: String? = "",
    val label: String? = "",
    val isPrimary: Boolean = false,
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "formattedAddress" to formattedAddress,
            "city" to city,
            "country" to country,
            "neighborhood" to neighborhood,
            "poBox" to poBox,
            "postCode" to postCode,
            "region" to region,
            "street" to street,
            "label" to label,
            "isPrimary" to isPrimary,
        )
    }
}

File: lib/contacts/domain/models/postal_address.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:one/utils/utils.dart';

part 'postal_address.freezed.dart';
part 'postal_address.g.dart';

@freezed
abstract class PostalAddress with _$PostalAddress {
  const factory PostalAddress({
    @Default('') String formattedAddress,
    @Default('') String city,
    @Default('') String country,
    @Default('') String neighborhood,
    @Default('') String poBox,
    @Default('') String postCode,
    @Default('') String region,
    @Default('') String street,
    @Default('') String label,
    @Default(false) bool isPrimary,
  }) = _PostalAddress;

  factory PostalAddress.fromJson(JsonMap json) => _$PostalAddressFromJson(json);
}
